File: frontend/src/types/bgi.ts
export interface BgiNode {
  id: string
  name: string
  children?: BgiNode[]
}

export interface BgiHierarchy {
  root: BgiNode | null
}

export interface BgiCloudLeadUser {
  id: string
  name: string
  email: string
  department: string
  team?: string
  initials: string
  role: string[]
  bgi_ids: string[]
}

File: backend/app/schemas/bgi.py
from typing import Any

from pydantic import BaseModel


class BgiNode(BaseModel):
    id: str
    name: str
    children: list["BgiNode"] | None = None


class BgiHierarchy(BaseModel):
    root: BgiNode | None = None


class BgiAssignProjectsRequest(BaseModel):
    bgi_id: str
    project_ids: list[str]


class BgiUnassignProjectsRequest(BaseModel):
    project_ids: list[str]
